python/src/lib.rs

```diff
@@ -42,6 +43,20 @@ struct RawDeltaTable {
     _table: deltalake::DeltaTable,
 }
 
+#[pyclass]
+struct RawDeltaTableMetaData {
```

Comment:
Curious what's the reason for wrapping this struct in python instead of just exposing it as `Metadata`. Are you concerned about extensibility from python side? If so, we could subclass `Metadata` with `RawDeltaTableMetaData`, that why, we won't need to copy all the attributes again in the `__init__` method.

Reply:
Good question!

I started by working directly with the `Metadata` structure without having a Python class. And I didn't feel comfortable manipulating a Python object without knowing anything about its structure (documentation, debug, attributes & methods). So, I changed my mind by creating the Python class and I was wondering if it might also be interesting to add the documentation on the Python properties also.